rdtools/soiling.py

```diff
@@ -343,6 +351,17 @@ def _calc_monte(self, monte, method='half_norm_clean'):
               the interval.
         '''
 
+        # Raise a warning if there is >20% invalid data
+        if (method == 'half_norm_clean') or (method == 'random_clean'):
+            valid_fraction = self.analyzed_daily_df['valid'].sum()/len(self.analyzed_daily_df)
```

Comment:
`valid_fraction = self.analyzed_daily_df['valid'].mean()` may be easier to read, up to you